File: python/challenge02/steps_to_exit.py
def steps_to_exit(nums: int) -> int:

    pos = 0
    steps = 0
    n = len(nums)

    while (pos >= 0 and pos < n):
        salto = nums[pos]
        nums[pos] += 1
        pos += salto
        steps += 1
    
    return steps

def main():
    total_steps = 0
    steps_last_line = 0

    with open("trace.txt", "r") as file:
        for line in file:
            l = line.strip()
            if not line:
                continue

            nums = [int(x) for x in l.split()]
            steps = steps_to_exit(nums)

            total_steps += steps
            steps_last_line = steps

    print(f"Submit: {total_steps}-{steps_last_line}")

if __name__ == "__main__":
    main()

File: node/challenge02/steps_to_exit.js
const fs = require('fs');
const readline = require('readline');

function stepsToExit(nums) {
    let pos = 0;
    let steps = 0;
    let n = nums.length;

    while (pos >= 0 && pos < n) {
        let salto = nums[pos];
        nums[pos] += 1;
        pos += salto;
        steps += 1;
    }

    return steps;
}

async function main() {
    const fileStream = fs.createReadStream('trace.txt', 'utf8');

    const rl = readline.createInterface({
        input: fileStream,
        crlfDelay: Infinity
    });

    let totalSteps = 0;
    let stepsLastLine = 0;

    for await (const line of rl) {
        const l = line.trim();
        if (l === '') continue;

        let nums = l.split(' ').map(Number);

        let steps = stepsToExit(nums);

        totalSteps += steps;
        stepsLastLine = steps;

    }
    console.log(`submit ${totalSteps}-${stepsLastLine}`);
}

main().catch(err => {
    console.error('Error:', err);
    process.exit(1);
});